components/BottomNavigation/src/MDCBottomNavigationBar.m

```diff
@@ -405,7 +405,7 @@ - (void)didCancelTouchesForButton:(UIButton *)button {
 #pragma mark - Setters
 
 - (void)setItems:(NSArray<UITabBarItem *> *)items {
-  if (_items == items) {
+  if ([_items isEqual:items]) {
```

Comment:
You likely want both so the const and nil case are handled:
if (items == _items || [_items isEqual:items])